src/interface.cpp

```diff
@@ -381,4 +467,370 @@ uint32_t ddwaf_builder_get_config_paths(
 }
 
 void ddwaf_builder_destroy(ddwaf_builder builder) { delete builder; }
+
+ddwaf_object *ddwaf_object_invalid(ddwaf_object *object)
+{
+    if (object == nullptr) {
+        return nullptr;
+    }
+
+    *object = {nullptr, 0, {nullptr}, 0, DDWAF_OBJ_INVALID};
+
+    return object;
+}
+
+ddwaf_object *ddwaf_object_null(ddwaf_object *object)
+{
+    if (object == nullptr) {
+        return nullptr;
+    }
+
+    *object = {nullptr, 0, {nullptr}, 0, DDWAF_OBJ_NULL};
+
+    return object;
+}
+
+ddwaf_object *ddwaf_object_string(ddwaf_object *object, const char *string)
+{
+    if (object == nullptr) {
+        return nullptr;
+    }
+
+    if (string == nullptr) {
+        DDWAF_DEBUG("tried to create a string from a nullptr pointer");
+        return nullptr;
+    }
+    return ddwaf_object_string_helper(object, string, strlen(string));
+}
+
+ddwaf_object *ddwaf_object_stringl(ddwaf_object *object, const char *string, size_t length)
+{
+    if (object == nullptr) {
+        return nullptr;
+    }
+
+    if (string == nullptr) {
+        DDWAF_DEBUG("Tried to create a string from a nullptr pointer");
+        return nullptr;
+    }
+
+    // NOLINTNEXTLINE(clang-analyzer-unix.Malloc)
+    return ddwaf_object_string_helper(object, string, length);
+}
+
+ddwaf_object *ddwaf_object_stringl_nc(ddwaf_object *object, const char *string, size_t length)
+{
+    if (object == nullptr) {
+        return nullptr;
+    }
+
+    if (string == nullptr) {
+        DDWAF_DEBUG("Tried to create a string from an nullptr pointer");
+        return nullptr;
+    }
+
+    *object = {nullptr, 0, {string}, length, DDWAF_OBJ_STRING};
+
+    return object;
+}
+
+ddwaf_object *ddwaf_object_string_from_signed(ddwaf_object *object, int64_t value)
+{
+    if (object == nullptr) {
+        return nullptr;
+    }
+
+    // INT64_MIN is 20 char long
+    char container[UINT64_CHARS] = {0};
+    const auto length = static_cast<std::size_t>(
+        // NOLINTNEXTLINE(cppcoreguidelines-pro-type-vararg,hicpp-vararg)
+        snprintf(container, sizeof(container), "%" PRId64, value));
+
+    return ddwaf_object_stringl(object, container, length);
+}
+
+ddwaf_object *ddwaf_object_string_from_unsigned(ddwaf_object *object, uint64_t value)
+{
+    if (object == nullptr) {
+        return nullptr;
+    }
+
+    // UINT64_MAX is 20 char long
+    char container[UINT64_CHARS] = {0};
+
+    const auto length = static_cast<size_t>(
+        // NOLINTNEXTLINE(cppcoreguidelines-pro-type-vararg,hicpp-vararg)
+        snprintf(container, sizeof(container), "%" PRIu64, value));
```

Comment:
you can use the more efficient `to_chars`. same in the signed version

Reply:
I'm not touching this code, just moved it over from one file to another. It'll be nuked.